### DISCUSSION TITLE
Rewrite the validator to use objects not strings.

### DISCUSSION THREAD

**Discussion by Jameron** [laravel/framework#43927] [Ideas] [answered]
The Laravel validation rules are written as strings. If they were instead written as objects you would prevent typos in validation rules from breaking at run time. Typos would instead be caught by static analyzers like PHAN and modern IDEs. 

Cake PHP has an excellent validation pattern that uses objects instead of strings, it should be referenced in the re-write / update.

Psuedocode:

Instead of `$validated = $request->validate([
        'title' => 'required|unique:posts|max:255',
        'body' => 'required',
    ]);`
 it could be

`$validated = $request->validate()
    ->string('title')
    ->required('title')
    ->unique('title', 'posts')
    ->maxLength(255)`

**Comment by chuckrincon** [comment, marked as answer]
This adds more complexity to the codebase rather than a nice DX to the user.
Though on the user land packages exists [Laravel Hyrule](https://github.com/square/laravel-hyrule), an object-oriented, fluent API for building validation rules.
